main.go

```diff
@@ -57,7 +58,7 @@ func (e Env) GetSecret(key string) string {
 	}
 	// Ideally environment above is set to avoid costly ssm (parameter store) lookups
 
-	ps := ssm.New(e.Cfg)
+	ps := ssm.New(thisEnvironment.Cfg)
```

Comment:
Since Go is strongly typed, if you didn't know what `e` was, you would hover your selector over it and it would tell it's about the Environment. You don't to shove the type in the variable name.